### ISSUE TITLE
[BUG] Cheetah 1.1b SD-SPI / Stepper-ENA conflict?

### ISSUE THREAD

**Issue by alfonsg00** [MarlinFirmware/Marlin#16904]
I have a issue with Marlin 2.0.3. I'm using an Ender 3 with the cheetah 1.1b board. If i remove and insert the sd card before starting a print the printhead acts strange and won't print properly. You can hear a "clonk" after re-inserting the sd card from the z axis. If you initiate eeprom before printing it will print normally. Tried to use EEPROM_AUTO_INIT but didn't make any difference.

**Comment by alfonsg00** [comment]
Link for videos of the issue: https://youtu.be/didVrh2fIPc
https://youtu.be/KQLpTWLgGNE

**Comment by andysport** [comment]
I have the same printer, board and marlin version too. Yes, I can confirm that bug. When I insert the SD, I can hear a noise and the stepper drives move with high speed and noise. If I restore failsafe, the issues disappears.

**Comment by Wotovski** [comment]
I have the same problem. I can't start any Printjob from my SD Cards. My Printer is a Anycubic Kossel.

**Comment by Hagbard-84** [comment]
I have the same problem. Ender 3, cheetah 1.1, Marlin 2.0.3.


---> **solved**: befor this happened the printer reported "no media" "media inserted".<--- it was the sd card slot detector pin Glitch.

**Comment by trouch** [comment]
Issue still is happening with current bugfix-2.0.x branch and doesn't seem to be related to the previous SD detect bug. 
I also have Fysetc Cheetah 1.1 board (STM32F1 + TMC2209/UART) with Flash EEPROM enabled by the way.

It is easy to reproduce by inserting the SD _after_ powering on.
Steppers/drivers configuration get somehow corrupted, X axis becomes super fast and loud, Y super slow and Homing fails then.

There is no issue when inserting the card _before_ powering on the printer.
Workaround is power-cycling the printer after SD card insertion, but it is not convenient.

I tried to debug but so far unlucky.

**Comment by trouch** [comment]
More testing ...

I've set `#define SDCARD_CONNECTION ONBOARD` in order to disable card detect and auto-mount, and issue is still is occurring.
As @alfonsg00 said, we can clearly hear a _clonk_ noise - motors being disabled - at the moment we insert the card, regardless of `SDCARD_CONNECTION` and auto-mount.

At this point I highly suspect an hardware issue or STM32(F1) specific one which break something when SPI lines are hotplugged.

Edited: 
Disabling `SDSUPPORT` doesn't make any change. Hot-plugging the SD still corrupt something.

Edited2:
I suspected the SD detect pin to conflict with USART2_CK, so I tested using Software Serial, but same bug.
So I used a small piece of tape onto the SD in order to trigger SD detect pin without SPI lines connected, and at this point the issue disappeared. Well I can't use SD obviously but TMC2209 doesn't get corrupted anymore when inserting the card 😅 

I really think about an hardware design issue since only Cheetah users seem affected.
